pkg/backends/cloudwatch/cloudwatch.go

```diff
@@ -142,6 +144,16 @@ func (client Client) buildMetricData(metrics *gostatsd.MetricMap) (metricData []
 		for _, pct := range timer.Percentiles {
 			addMetricData(key+"."+pct.Str, "Milliseconds", pct.Float, timer.Tags)
 		}
+		if timer.Histogram != nil {
```

Comment:
@tmszdmsk if we wanted to enable the ability to send both the histogram and the original timer metrics, would it be as simple as changing this logic slightly to drop the histogram buckets tag for those timer metrics?

Reply:
Sending both through would be bad, because it makes it impossible to not send both. If both are desired (for example, doing side by side testing) then the client should be sending both - ideally with different names so as not to pollute the namespace in the event that the wrong tags make it out to a test gostatsd deployment.

Reply:
Could we control that behaviour with a configuration flag? The UX isn't really that great if we're asking users to duplicate code for sending metrics anywhere they want a histogram.

> ideally with different names so as not to pollute the namespace in the event that the wrong tags make it out to a test gostatsd deployment

wouldn't this be a non-issue given we append `.histogram` to those metrics already?

Reply:
I haven't looked at the code for a while, yeah - if we're adding a suffix, pollution isn't an issue.

The primary issue really is the inability to not send both. Lack of control in what is emitted is a large cause of high cardinality (ie, timer aggregations).

I'd personally make the argument that work is required to move to the new style anyway, both in what is emitted (adding buckets) and querying the data, so there's going to be code change regardless.

Reply:
Alright, that makes more sense. We'll leave it as is.